1Password/developer-community-projects
# [DEV] Augment and Validate Project Date Data (Breaking Change)

## Summary

Every project within [projects.json](https://github.com/1Password/developer-community-projects/blob/main/projects.json) that has a `category` of 'repo' has been given a date field called `createdAt`. Projects that had a field in the past called `date` have now been replaced with this new property: `createdAt` (breaking change) 

## Related Issues and MRs

Closing #23 

## What changed

- Add a date for each project that has a category of "repo"
- Update `Project` struct in `validate.go`. Date field is now required for all project categories.
- `date` has become `createdAt` (breaking change)
- Update `CONTRIBUTING.md` with data shape information.

## Steps for reviewer

1. Pull changes
```
git checkout brian/date-validation
```
2. Set the `currentBranch` value on line 31 of `validate.go` to "main"
3. Run the validator on `projects.json`:
```
go run validate.go
```
4. Ensure all projects have a date field called `createdAt`
5. Ensure the meta data of the `CreatedAt` field is accurate within `validate.go`

## Additional resources

N/A
